hr_timesheet_auto_creation/models/hr_timesheet_sheet.py

```diff
@@ -30,41 +30,17 @@ def create_employee_timesheet(self):
             lambda x: x.employee_id.id, exists_timesheet_records
         )
         employee_ids = list(set(employee_ids) - set(ignore_employee_ids))
-        vals = {
-            "date_start": monday,
-            "date_end": sunday,
-        }
         for employee_id in employee_ids:
-            vals["employee_id"] = employee_id
-            self.sudo().create(vals)
+            self.sudo().create(
+                {
+                    "employee_id": employee_id,
+                    "date_start": monday,
+                    "date_end": sunday,
+                }
```

Comment:
I don't understand the difference between 
`vals["employee_id"] = employee_id` and `{"employee_id": employee_id}`

Reply:
`create()` modifies the provided dict. the `create()` function of `mail.thread` contains this:
```python
# subscribe uid unless asked not to
if not self._context.get('mail_create_nosubscribe'):
    for values in vals_list:
        message_follower_ids = values.get('message_follower_ids') or []
        message_follower_ids += [(0, 0, fol_vals) for fol_vals in self.env['mail.followers']._add_default_followers(self._name, [], self.env.user.partner_id.ids, customer_ids=[])[0][0]]
        values['message_follower_ids'] = message_follower_ids
```
after the call, the provided dict contains a list of followers with `"message_follower_ids"` as key.

in the first case, it is the same dict that is reused between consecutive `create()` calls. because of this, at the next call, the same follower will be added to the list because it already exists.

in the second case, a new dict is created and used at each loop iteration, so even though the dict is modified, it works as expected.

Reply:
thank you for the cristal clear explanation :) I had no clue `create()` would modify the `dict`

Reply:
i had no clue either. it was a nasty surprise. although, when i first saw the same dict being reused between loop iterations, just changing one value, it smelled not very good. unfortunately, i did not follow that track to debug, or it would have been much quicker to find.

i don’t know whether the fact that `create()` modifies the dict is documented somewhere.